test/e2e/serial/tests/workload_unschedulable.go

```diff
@@ -289,40 +289,21 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload unsched
 		})
 
 		It("[test_id:47619][tier3][unsched][default-scheduler] a deployment with a guaranteed pod resources available on one node but not on a single numa; scheduled by default scheduler", func() {
+			By("Scheduling the testing pod")
```

Comment:
the code you are adding fixes the issue at hand, but isn't the test trying to exercise exactly this case?
IOW, in this case we do want exactly a Deployment to trigger this scenario (if memory serves).

What the test is supposed to do is to check at any random time that the pods pertaining to this deployment are failing as expected.
This alone should not drag the cluster down though.

Additionally, this test didn't cause any major issues in my own test runs, so it would be nice to understand why is failing/flaking.

Reply:
The test indeed wanted to test the scenario on deployment/replicaset. 

> What the test is supposed to do is to check at any random time that the pods pertaining to this deployment are failing as expected.
This alone should not drag the cluster down though.

I agree on this except for the term random time. The automation will check all the deployment's pods they hit TAE, but in case the pod was checked in a time that the TAE event wasn't there yet the test would fail. 
this led me to simplify the test to check a single pod and adding the Eventually part. 
However thinking about it again now (and testing it) I could have added the eventually without simplifying the test, so that the event check will be performed on each pod in an Eventually. 
ran the test with the new fix couple of times and it seems stable now.

Reply:
However, note that we fetch the deployment's pods only once so the test will check only a sample of the deployment's pods, which is enough and makes the test on the safe side than getting stuck in a loop of checking new pods every time.

Reply:
thanks for pointing this out!
the code is updated now